### ISSUE TITLE
Feature request: fallback

### ISSUE THREAD

**Issue by miniBill** [andreasewering/travelm-agency#51]
Would it be possible to mark one language as the "fallback" language, and get all the missing strings from that file?

**Comment by andreasewering** [comment]
Yeah that would certainly be possible. What should the usage look like? 

I could see a command-line argument like --default_language=en for example.

The use case would be that some language do not have translations for all keys, correct?

**Comment by miniBill** [comment]
Yes, especially for hobby projects, I think it would be common to have volunteer translate, so having missing strings would be probable.

A command line argument like you describe seems absolutely reasonable. A nicer option would be a per-file default: `main.en-UK.ftl` has on its first line something like `# fallback-language: en-US`

**Comment by andreasewering** [comment]
I like the per-file default. Will implement that in the next week or so :)

**Comment by andreasewering** [comment]
I just noticed that while .properties and .ftl files allow for comments like these, .json files cannot (or at least IDE support is confused even if I could technically parse some kind of top-level comment syntax in json.

Maybe I'll add a reserved "_fallback-language" key or something. Not sure yet, suggestions are welcome.

**Comment by miniBill** [comment]
A reserved key is inelegant, but given JSON stubborn refusal of comments, I think it's the most sensible option. You might have the key be something that would not generate valid Elm identifiers? Like, I dunno `"--fallback-language"`?

I'm going to use the Fluent format only, so I don't really have a horse in this race

**Comment by andreasewering** [comment]
I implemented the fallback behaviour for all three file types.
In inline mode, the code generator simply delegates onto the given fallback accessor, i.e.

```
de = 
  { normalKey = "abc"
  , fallbackKey = en.fallbackKey
  }

en = 
  { normalKey = "bcd"
  , fallbackKey = "this will also be shown in german"
```

which is pretty nice because the string will not appear 2 times in the JS bundle.

In dynamic mode, I went for the "dumb" approach, which just writes the respective value from the other language into the json file, essentially duplicating the fallbacked key/value pair. I think that's reasonable behaviour, but as always, I'm willing to take suggestions :)

I still need to add a few tests and add back the compile time key completeness checking (which was previously happening after a translation file is loaded but now has to be deferred to the last translation file is loaded to avoid errors dependant on ordering).
